src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs

```diff
@@ -154,12 +154,18 @@ public int Capacity
 
             set
             {
-                if ((uint)index >= (uint)_size)
+                T[] array = _items;
+
+                // doing the array boundry check on our own here improves the perf by 10%
+                if ((uint)index < (uint)_size && (uint)index < (uint)array.Length)
```

Comment:
Presumably the second redundant check is to trigger an optimization by the JIT?

Reply:
yes, that is the trick. It's quite common in the `List<T>` : https://github.com/dotnet/runtime/blob/a23cc775e21112797eecbb23a72769063849e1b3/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs#L201

the other part of the trick is to cast to `uint` to avoid two comparisons (`index > 0 && index < length`)

Reply:
If so, can you clarify that this redundancy is intentional in the comment above?